pkg/scheduler/clustereligibilitychecker/checker.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package clustereligibilitychecker features a utility for verifying if a member cluster is
+// eligible for resource placement.
+package clustereligibilitychecker
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	// defaultClusterHeartbeatTimeout is the default timeout value this checker uses for checking
+	// if a cluster has been disconnected from the fleet for a prolonged period of time.
+	defaultClusterHeartbeatTimeout = time.Minute * 5
+
+	// defaultClusterHealthCheckTimeout is the default timeout value this checker uses for checking
+	// if a cluster is still in a healthy state.
+	defaultClusterHealthCheckTimeout = time.Minute * 5
+)
+
+type ClusterEligibilityChecker struct {
+	// The timeout value this checker uses for checking if a cluster has been disconnected
+	// from the fleet for a prolonged period of time.
+	clusterHeartbeatTimeout time.Duration
```

Comment:
```suggestion
	clusterHeartbeatCheckTimeout time.Duration
```

niiiit, make it consistent with "healthCheckTimeout"?

Reply:
Done.